src/vsr/replica.zig

```diff
@@ -3656,6 +3696,10 @@ pub fn ReplicaType(
                     self.primary_pipeline_prepare(request);
                 }
 
+                if (self.pulse_needed()) {
+                    self.send_request_pulse_to_self();
+                }
```

Comment:
```suggestion
                if (self.pulse_needed()) {
                    assert(self.upgrade_release == null);
                    self.send_request_pulse_to_self();
                }
```
(To be symmetrical :smile:)